rust/benches/producer.rs

```diff
@@ -0,0 +1,168 @@
+            ip: IpAddr::V4(Ipv4Addr::LOCALHOST),
+            announced_ip: None,
+            port: None,
+            send_buffer_size: None,
+            recv_buffer_size: None,
+        }));
+
+    let transport_1 = router
+        .create_webrtc_transport(transport_options.clone())
+        .await
+        .expect("Failed to create transport1");
+
+    let transport_2 = router
+        .create_webrtc_transport(transport_options)
+        .await
+        .expect("Failed to create transport2");
+
+    (worker, router, transport_1, transport_2)
+}
+
+fn audio_producer_options() -> ProducerOptions {
+    let options = ProducerOptions::new(
+        MediaKind::Audio,
+        RtpParameters {
+            mid: Some(fastrand::u32(100_000_000..999_999_999).to_string()),
+            codecs: vec![RtpCodecParameters::Audio {
+                mime_type: MimeTypeAudio::Opus,
+                payload_type: 0,
+                clock_rate: NonZeroU32::new(48000).unwrap(),
+                channels: NonZeroU8::new(2).unwrap(),
+                parameters: RtpCodecParametersParameters::from([
+                    ("useinbandfec", 1_u32.into()),
+                    ("usedtx", 1_u32.into()),
+                    ("foo", "222.222".into()),
+                    ("bar", "333".into()),
+                ]),
+                rtcp_feedback: vec![],
+            }],
+            header_extensions: vec![
+                RtpHeaderExtensionParameters {
+                    uri: RtpHeaderExtensionUri::Mid,
+                    id: 10,
+                    encrypt: false,
+                },
+                RtpHeaderExtensionParameters {
+                    uri: RtpHeaderExtensionUri::AudioLevel,
+                    id: 12,
+                    encrypt: false,
+                },
+            ],
+            // Missing encodings on purpose.
+            encodings: vec![],
+            rtcp: RtcpParameters {
+                cname: Some("audio-1".to_string()),
+                ..RtcpParameters::default()
+            },
+        },
+    );
+
+    options
+}
+
+pub fn criterion_benchmark(c: &mut Criterion) {
+    let mut group = c.benchmark_group("producer");
+
+    {
+        let (_worker, _router, transport_1, _transport_2) =
+            futures_lite::future::block_on(async { init().await });
+
+        let producer = futures_lite::future::block_on(async {
+            let (_worker, _router, transport_1, _transport_2) = init().await;
+            transport_1
+                .produce(audio_producer_options())
+                .await
+                .expect("Failed to produce audio")
+        });
+
+        group.bench_function("create", |b| {
+            b.iter(|| {
+                let _ = futures_lite::future::block_on(async {
+                    transport_1
+                        .produce(audio_producer_options())
+                        .await
+                        .expect("Failed to produce audio")
+                });
```

Comment:
You are not only testing creation here, you also trigger destruction by dropping the value here. Destruction is async though, so I guess the bench is fine for comparison reasons.